packages/sdk/src/oracle/optimistic/store/write.ts

```diff
@@ -87,7 +87,7 @@ export class OptimisticOracle {
     }
   }
 }
-export class Chain {
+export class Chain implements types.interfaces.Chain<state.OptimisticOracleEvent> {
```

Comment:
I want to note a design choice here.

The way you're approaching this seems to be: keeping all the classes backwards compatible, but having them inherit from newly-generic interfaces. When we add the Skinny OO client, it will be the same, but it will need its own set of clients that inherit from these generic interfaces, but pass in different generic arguments to those interfaces. The code will mostly be the same outside of a few instances where the logic interacts with the parts of the types that are different.

The alternative design is to make the client itself generic and branch within the client where necessary. This would be less overall code, but it probably wouldn't be backwards compatible (since the generic types probably wouldn't be inferred and you might require additional user inputs), and it might require more complex logic _within_ the client to deal with the two possible contract types.

Do you agree with this analysis?